project/frontend/src/components/RestaurantCardDeck.js

```diff
@@ -0,0 +1,127 @@
+import './RestaurantCardDeck.css';
+
+import React, { useReducer } from 'react';
+
+import RestaurantCard from './RestaurantCard.js';
+
+/**
+ * Types of deck actions.
+ * @enum {number}
+ */
+const Action = {
+  NOT_INTERESTED: -1,
+  NEUTRAL: 0,
+  INTERESTED: 1,
+  BACK: 2,
+};
+
+/**
+ * A reducer for the RestaurantCardDeck component. Updates RestaurantCardDeck
+ * state based on the previous state and a specified action.
+ *
+ * @param {!Object<string, *>} previous The previous state.
+ * @param {!Action} action An action type that may determine how the
+ *     state is updated.
+ */
+function reducer(previous, action) {
+  let index = previous.index;
+  const history = previous.history;
+
+  switch (action.type) {
+    case Action.INTERESTED:
+    case Action.NOT_INTERESTED:
+    case Action.NEUTRAL:
+      history[index] = action.type;
+      index++;
+      break;
+    default:
```

Comment:
Lets add the back action explicitly as a case as well and it can fall through to default 

case Action.BACK:
default:
     if index > 0: index--